include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -538,14 +537,92 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
                 dealii::ExcMessage("Not implemented."));
   }
 
-
   // TURBULENCE
-  if(use_turbulence_model)
+  if(viscosity_model_data.use_turbulence_model)
   {
-    AssertThrow(turbulence_model != TurbulenceEddyViscosityModel::Undefined,
-                dealii::ExcMessage("parameter must be defined"));
-    AssertThrow(turbulence_model_constant > 0,
-                dealii::ExcMessage("parameter must be greater than zero"));
+    AssertThrow(viscosity_model_data.turbulence_model_data.turbulence_model !=
+                  TurbulenceEddyViscosityModel::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+    AssertThrow(viscosity_model_data.turbulence_model_data.constant > 0,
+                dealii::ExcMessage("Parameter must be greater than zero."));
+    AssertThrow(viscosity_model_data.treatment_of_nonlinear_viscosity !=
+                  TreatmentOfNonlinearViscosity::Undefined,
+                dealii::ExcMessage("Parameter must be defined."));
+
+    if(temporal_discretization == TemporalDiscretization::BDFDualSplittingScheme)
+    {
+      AssertThrow(viscosity_model_data.treatment_of_nonlinear_viscosity !=
```

Comment:
Writing the assert in terms of what is currently supported could be safer for the future, e.g. if further options that are also not supported by the dual splitting scheme are added in the future, this assert does not have to be adjusted.